torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -1,5 +1,7 @@
 #include <torch/csrc/distributed/rpc/tensorpipe_agent.h>
 
+#ifdef USE_TENSORPIPE
+
```

Comment:
Rather than having #ifdef guard, wouldn't it be better to exclude the file from the list of sources in torch/CMakeLists.txt?

Reply:
I thought it would simpler (and more robust?) to add this guard in one place than to remove a file from the sources across three different build systems...

Reply:
You have to modify build system anyway, to add dependent libraries and so on. Also, making a typo in those build definitions leaves the test disabled forever, which is very hard to find compared to compile/link errors.
If you insist of #ifdef guards, wouldn't it be better to make them negative (i.e. compile the code unless `DISABLE_TENSORPIPE` define is set)